docs/keycloak.md

```diff
@@ -0,0 +1,356 @@
+# Keycloak Configuration
+
+Users are registered on Keycloak, and thanks to it are authenticated and authorized
+to perform actions against the APIs provided by Orchestra Cities. The following
+section describes how Keycloak is configured and utilised to accomplish this,
+alongside the Administration Portal.
+
+## Security Configuration
+
+![Keycloak and Portal diagram](rsrc/keycloak/portal_keycloak_diagram.png)
+
+Keycloak is used to both authenticate and authorise users who wish to access the
+Orchestra Cities resources through the APIs. With the Portal for instance, we have
+two clients used: the resource-server client, and the portal client.
+The portal is deployed as a backend (providing its own API), and a frontend that users
+can access through a web browser. The frontend authenticates users via the portal
+client, and the backend authorises users via the resource-server client.
+
+## Keycloak Clients
+
+The portal client in Keycloak is set up as public, meaning it doesn't
+require a secret to access. This client is used by the portal to authenticate users,
+like those logging into the Portal's frontend. Being public, it is not configured to
+give control over client scopes and roles.
+
+![Keycloak clients](rsrc/keycloak/keycloak_clients.png)
+
+The resource-server client is confidential, meaning that its client-secret
```

Comment:
`resource_server` (also use the "code" to evidence)

Reply:
list also the other clients and what they are for, also in describing state if they have a specific configuration, e.g. need roles (e.g. urbo as `admin` and `user`). and the default scope assigned `urbo:roles`

Reply:
not sure this was covered @Cerfoglg